superset/migrations/versions/978245563a02_migrate_iframe_to_dash_markdown.py

```diff
@@ -0,0 +1,170 @@
+
+from superset import db
+
+# revision identifiers, used by Alembic.
+revision = "978245563a02"
+down_revision = "f2672aa8350a"
+
+Base = declarative_base()
+
+
+class Slice(Base):
+    """Declarative class to do query in upgrade"""
+
+    __tablename__ = "slices"
+    id = Column(Integer, primary_key=True)
+    params = Column(Text)
+    viz_type = Column(String(250))
+
+
+dashboard_slices = Table(
+    "dashboard_slices",
+    Base.metadata,
+    Column("id", Integer, primary_key=True),
+    Column("dashboard_id", Integer, ForeignKey("dashboards.id")),
+    Column("slice_id", Integer, ForeignKey("slices.id")),
+)
+
+
+class Dashboard(Base):
+    __tablename__ = "dashboards"
+    id = Column(Integer, primary_key=True)
+    position_json = Column(Text)
+    slices = relationship("Slice", secondary=dashboard_slices, backref="dashboards")
+
+
+def create_new_markdown_component(chart_position, url):
+    return {
+        "type": "MARKDOWN",
+        "id": "MARKDOWN-{}".format(uuid.uuid4().hex[:8]),
+        "children": [],
+        "parents": chart_position["parents"],
+        "meta": {
+            "width": chart_position["meta"]["width"],
+            "height": chart_position["meta"]["height"],
+            "code": f'<iframe src="{url}" width="100%" height="100%"></iframe>',
+        },
+    }
+
+
+def upgrade():
+    bind = op.get_bind()
+    session = db.Session(bind=bind)
+
+    dash_to_migrate = defaultdict(list)
+    iframe_urls = defaultdict(list)
+
+    try:
+        # find iframe viz_type and its url
+        iframes = session.query(Slice).filter_by(viz_type="iframe").all()
+        iframe_ids = [slc.id for slc in iframes]
+
+        for iframe in iframes:
+            iframe_params = json.loads(iframe.params or "{}")
+            url = iframe_params.get("url")
+            iframe_urls[iframe.id] = url
+
+        # find iframe viz_type that used in dashboard
+        dash_slc = (
+            session.query(dashboard_slices)
+            .filter(dashboard_slices.c.slice_id.in_(iframe_ids))
+            .all()
+        )
+        for entry in dash_slc:
+            dash_to_migrate[entry.dashboard_id].append(entry.slice_id)
+        dashboard_ids = list(dash_to_migrate.keys())
+
+        # replace iframe in dashboard metadata
+        dashboards = (
+            session.query(Dashboard).filter(Dashboard.id.in_(dashboard_ids)).all()
+        )
+        for i, dashboard in enumerate(dashboards):
+            print(
+                f"scanning dashboard ({i + 1}/{len(dashboards)}) dashboard: {dashboard.id} >>>>"
+            )
+
+            # remove iframe slices from dashboard
+            dashboard.slices = [
+                slc for slc in dashboard.slices if slc.id not in iframe_ids
+            ]
+
+            # find iframe chart position in metadata
+            # and replace it with markdown component
+            position_dict = json.loads(dashboard.position_json or "{}")
+            for key, value in position_dict.items():
```

Comment:
nit: replace `value` with `chart_position` (or perhaps `position_value` as it's more clear

Reply:
i like `chart_position`. fixed.